sherpa/plot/backends.py

```diff
@@ -0,0 +1,538 @@
+#
```

Comment:
I can't really add a comment to a particular commit - in this case b40f0ddd800df8c22bba3fd0b01bb2d222601698 - so I'm just adding it here. This commit removes `dummy_backend.py` which means that I believe Sherpa is unusable if you don't have a plotting backend. It looks like this doesn't get addressed for a few commits. Ideally we would have each commit be able to pass CI (e.g. for bisect) but I know we have not required this in the past.

Reply:
I have not programmed like that. If we want that, the only thing I can do that this point is to squash all commits into one. I knew I was changing a whole bunch of stuff and it was just not practical (if even possible) to do all the code changes and all the test changes at the same time, while also keeping commits not too big.

Should I squash all commits into one?

Reply:
Well, in a number of my PRs I've done things like move commits around, then squash some of them, and then broke them up again where I felt it lead to smaller individual commits, and lets me check individual commits still pass CI (or at least on my machine, I don't always get a CI run, particularly after the nth rebase/rework). It's a lot of work to do so, and has the potential to mess things up (so I often have several copies of the repository lying around just in case), but it can lead to a better understanding of how the code fits together (in particular for long-running PRs like this one).

I feel that this PR could be reduced in the number of commits, but whether it would be down to e.g. two (e.g. all the code changes then all the RTD changes), or whether it makes sense to break up the code changes a bit more I don't have a feel for. I also think that it's not worth worrying about just yet for the code as you are still working on it (the RTD changes could probably be squashed into one)

Reply:
I'm happy to squash, but I really, really don't want to break that up again into more commits after that. As you say, it's a lot of work, it can easily mess up stuff, and it's of questionable benefit (read: the boundaries where to split are subjective). But I will keep this request in mind as I continue to work on Sherpa in the future and try to only make clean commits that pass CI. 
(Other other big project I'm involved in does not care if CI passes on intermediate commits, as long as the commit message is useful, so I was just not aware that that's expected as part of the Sherpa workflow. Sorry; I'll try to improve in the future.)